apps/contribution/test/contribution.js

```diff
@@ -0,0 +1,231 @@
+      0x0,
+      false,
+      { from: root }
+    );
+    contributor = Contributor.at(
+      receipt.logs.filter(l => l.event == 'NewAppProxy')[0].args.proxy
+    );
+
+    //init app
+    await contribution.initialize(appsId);
+
+    await acl.createPermission(
+      root,
+      contribution.address,
+      await contribution.ADD_CONTRIBUTION_ROLE(),
+      root,
+      { from: root }
+    );
+
+    await acl.createPermission(
+      root,
+      contribution.address,
+      await contribution.VETO_CONTRIBUTION_ROLE(),
+      root,
+      { from: root }
+    );
+
+    //init token (app)
+    await token.initialize(appsId);
+
+    //create token mint permission for coin owner
+    await acl.createPermission(
+      contribution.address,
+      token.address,
+      await token.MINT_TOKEN_ROLE(),
+      root,
+      { from: root }
+    );
+
+    //init contributor app
+    await contributor.initialize(root, appsId);
+
+    await acl.createPermission(
+      root,
+      contributor.address,
+      await contributor.MANAGE_CONTRIBUTORS_ROLE(),
+      root,
+      { from: root }
+    );
+  });
+
+  describe("Owner default space permissions", async () => {
+    it('check owner can add contribution', async () => {
+      let addContributionPermission = await acl.hasPermission(root, contribution.address, await contribution.ADD_CONTRIBUTION_ROLE());
+      // eslint-disable-next-line no-undef
+      assert.equal(addContributionPermission, true);
+    });
+
+    it('check owner can veto contribution', async () => {
+      let vetoContributionPermission = await acl.hasPermission(root, contribution.address, await contribution.VETO_CONTRIBUTION_ROLE());
+      // eslint-disable-next-line no-undef
+      assert.equal(vetoContributionPermission, true);
+    });
+
+    it('check contribution app can mint token', async () => {
+      let mintTokenPermission = await acl.hasPermission(contribution.address, token.address, await token.MINT_TOKEN_ROLE());
+      // eslint-disable-next-line no-undef
+      assert.equal(mintTokenPermission, true);
+    });
+  });
+
+  describe("Add contribution", async () => {
+    let amount = 100;
+    let contributorId = 1;
+    let hashDigest = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    let hashFunction = 1;
+    let hashSize = 1;
+
+    it("should revert when add contribution from address that does not have permission", async () => {
+      return assertRevert(async () => {
+        await contribution.add(amount, contributorId, hashDigest, hashFunction, hashSize, {from: member1});
+        'sender does not have permission';
+      });
+    });
+
+    it("should revert when add contribution with amount equal to zero", async () => {
+      return assertRevert(async () => {
+        await contribution.add(0, contributorId, hashDigest, hashFunction, hashSize, {from: root});
+        'amount equal to zero';
+      });
+    });
+
+    it("add contribution", async () => {
```

Comment:
Looks like the expectation wasn't described here (`it("does x")`). It also looks like there are 3 different expectations in one block here.